src/Init/Data/String/Basic.lean

```diff
@@ -196,16 +196,48 @@ def utf8PrevAux : List Char → Pos → Pos → Pos
     let i' := i + c
     if i' = p then i else utf8PrevAux cs i' p
 
+/--
+Returns the position in a string before a specified position, `p`. If `p = ⟨0⟩`, returns `0`.
+If `p` is not a valid position, the result is unspecified.
+
+Examples:
+* `"abc".prev ⟨2⟩ = String.Pos.mk 1`
+* `"abc".prev ⟨0⟩ = String.Pos.mk 0`
+* `"L∃∀N".prev ⟨4⟩ = String.Pos.mk 1`, since `'∃'` is a multi-byte UTF-8 character
+-/
 @[extern "lean_string_utf8_prev"]
 def prev : (@& String) → (@& Pos) → Pos
   | ⟨s⟩, p => if p = 0 then 0 else utf8PrevAux s 0 p
 
+/--
+Returns the first character in `s`. If `s = ""`, returns `(default : Char)`.
+
+Examples:
+* `"abc".front = 'a'`
+* `"".front = (default : Char) = 'A'`
```

Comment:
This example is confusing, since you can't actually chain `=` like that in Lean!
```suggestion
* `"".front = (default : Char)`
```

Reply:
I think it even makes sense to leave off the `'A'` entirely here, because there's nothing really special about that being the default `Char` - we could just as easily make it be `'∃'` someday.

One argument for including it is that it could provide a clue to users who are getting some weird data, though.